spec/helpers.lua

```diff
@@ -3228,27 +3228,41 @@ local function prepare_prefix(prefix)
 end
 
 
---- Cleans the Kong environment.
--- Deletes the working directory if it exists.
--- @param prefix (optional) path to the working directory, if omitted the test
--- configuration will be used
--- @function clean_prefix
-local function clean_prefix(prefix)
-  prefix = prefix or conf.prefix
-  if pl_path.exists(prefix) then
-    local _, err = pl_dir.rmtree(prefix)
+local function clean_dir(dir)
+  if pl_path.exists(dir) then
+    local _, err = pl_dir.rmtree(dir)
     -- Note: gojira mount default kong prefix as a volume so itself can't
     -- be removed; only throw error if the prefix is indeed not empty
     if err then
-      local fcnt = #assert(pl_dir.getfiles(prefix))
-      local dcnt = #assert(pl_dir.getdirectories(prefix))
+      local fcnt = #assert(pl_dir.getfiles(dir))
+      local dcnt = #assert(pl_dir.getdirectories(dir))
       if fcnt + dcnt > 0 then
         error(err)
       end
     end
   end
 end
 
+local function get_runtime_data_path(prefix)
+  return utils.get_runtime_data_path(prefix)
+end
+
+local function clean_runtime_dir(prefix)
```

Comment:
shouldn't cleaning the runtime dir be implicit in `clean_prefix`? The names currently describe best what they really do, but this might be a breaking change for many plugin tests that will not know about the existence of the new `clean_runtime_dir`, hence those tests might start failing.

More so because the issue we had with the dangling unix sockets was intermittent and not consistent. So will cause some headaches with plugin developers.

Reply:
Was mainly concerned with the naming causing confusion but did adjust clean_prefix to include runtime_dir